### PR TITLE
JS Component Documentation

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

The idea is to lower the learning curve for developers by making it easy for them to leverage our existing components in their custom component. We do this by exporting the "Base" version of the component, along with any additional classes of note, so that they can be imported in another package. If we make our components easy to compose, it opens the door for really cool and complex components!

We also add a very minimal doc to each js package by adding or modifying the existing `README` (@aliabd if you can add these to the website it would be awesome 🙏 )

I tried to be consistent with the names. So most classes are prefixed with `Base` (following the lead of `BaseButton`) except the ones that were already being exported!!

- [x] SimpleVideo
- [x] SimpleAudio
- [x] Expose the `Base` component where it makes sense
- [x] Update the READMEs of the components

How to test `SimpleAudio` and `SimpleVideo`

Clone, install, and build this branch. Then run the following demos

```python

import gradio as gr
#from gradio_myvideo import MyVideo
from gradio._simple_templates import SimpleVideo as MyVideo

example = MyVideo().example_inputs()


with gr.Blocks() as demo:
    with gr.Row():
        with gr.Column():
            f1 = MyVideo()
        with gr.Column():
            f2 = MyVideo()
    with gr.Row():
        n_change_1 = gr.Number(0, label="change 1")
        n_change_2 = gr.Number(0, label="change 2")
        n_upload_1 = gr.Number(0, label="upload 1")
        stop_2 = gr.Number(0, label="Stop 2")
        play_2 = gr.Number(0, label="Play 2")
        pause_2 = gr.Number(0, label="Pause 2")

    f1.upload(lambda f, n_upload_1: (f, n_upload_1 + 1), [f1, n_upload_1], [f2, n_upload_1])
    f1.change(lambda n_change_1: n_change_1 + 1, [n_change_1], [n_change_1])
    f2.change(lambda n_change_2: n_change_2 + 1, [n_change_2], [n_change_2])

    f2.stop(lambda stop_2: stop_2 + 1, [stop_2], [stop_2])
    f2.play(lambda play_2: play_2 + 1, [play_2], [play_2])
    f2.pause(lambda pause_2: pause_2 + 1, [pause_2], [pause_2])

demo.launch()
```

The Audio play, pause, stop events don't seem to work on v4. Posted about it on slack today.
```python

import gradio as gr
from gradio._simple_templates import SimpleAudio as MyAudio

example = MyAudio().example_inputs()


with gr.Blocks() as demo:
    with gr.Row():
        with gr.Column():
            f1 = MyAudio()
        with gr.Column():
            f2 = MyAudio()
    with gr.Row():
        n_change_1 = gr.Number(0, label="change 1")
        n_change_2 = gr.Number(0, label="change 2")
        n_upload_1 = gr.Number(0, label="upload 1")
        stop_2 = gr.Number(0, label="Stop 2")
        play_2 = gr.Number(0, label="Play 2")
        pause_2 = gr.Number(0, label="Pause 2")

    f1.upload(lambda f, n_upload_1: (f, n_upload_1 + 1), [f1, n_upload_1], [f2, n_upload_1])
    f1.change(lambda n_change_1: n_change_1 + 1, [n_change_1], [n_change_1])
    f2.change(lambda n_change_2: n_change_2 + 1, [n_change_2], [n_change_2])

    f2.stop(lambda stop_2: stop_2 + 1, [stop_2], [stop_2])
    f2.play(lambda play_2: play_2 + 1, [play_2], [play_2])
    f2.pause(lambda pause_2: pause_2 + 1, [pause_2], [pause_2])

    gr.Examples([example], inputs=[f1])

demo.launch()
```

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: js/app/build_plugins.ts [hunk 1]
```diff
@@ -233,6 +233,8 @@ const ignore_list = [
 	"preview",
 	"simpledropdown",
 	"simpletextbox",
+	// "simplevideo",
```
Comment (freddyaboulton) [inline js/app/build_plugins.ts:236]: Will revert before merge. 

But in order to test these without doing a release, we need to build the app.

File: js/app/src/lite/ErrorDisplay.svelte [hunk 2]
```diff
@@ -21,20 +21,19 @@
...
-	{#if error.message}
-		<p class="error-name">
-			{error.message}
-		</p>
-	{/if}
-	{#if error.stack}
-		<pre class="error-stack"><code>{error.stack}</code></pre>
-	{/if}
-	{/if}
+		{#if error}
```
Comment (freddyaboulton) [inline js/app/src/lite/ErrorDisplay.svelte:24]: I ran `pnpm format:write` and it changed a bunch of stuff lol

Comment (freddyaboulton) [reply js/app/src/lite/ErrorDisplay.svelte:24]: Anything that's not 

* Modifying an Index.svelte or package.json to export some classes
* A new or changed README
* related to SimpleAudio or SimpleVideo

Is probably just automatic lint

File: js/video/shared/VideoPreview.svelte [hunk 3]
```diff
@@ -64,13 +65,15 @@
...
 	{/key}
 	<div class="icon-buttons" data-testid="download-div">
-		<a
-			href={value.path}
-			target={window.__is_colab__ ? "_blank" : null}
-			download={value.orig_name || value.path}
-		>
-			<IconButton Icon={Download} label="Download" />
-		</a>
+		{#if show_download_button}
```
Comment (freddyaboulton) [inline js/video/shared/VideoPreview.svelte:68]: Added this to control showing download button as an output component

**Review by pngwn — APPROVED** [review]

**Review by aliabd — APPROVED** [review]
Went through the all the readmes and made the changes. everything looks good!! Thanks @freddyaboulton take another look when website builds

**Comment by freddyaboulton** [comment]
Thanks for the review and help @aliabd @pngwn  !

**Pull request merged**